From 176cb8b221fe1cb309a55804753e311261982106 Mon Sep 17 00:00:00 2001
Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
Date: Fri, 6 Dec 2024 09:16:31 -0300
Subject: [PATCH] csm-observability: fix pull image error

This patch consists of adding imagePullSecrets with name
default-registry-key only in ServiceAccounts.
It's necessary because without it the system can't use the
images from the registry local.

Signed-off-by: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
---
 .../karavi-metrics-powerflex-service-account.yaml     |  2 ++
 .../karavi-metrics-powermax-service-account.yaml      |  2 ++
 .../karavi-metrics-powerscale-service-account.yaml    |  2 ++
 .../karavi-metrics-powerstore-service-account.yaml    |  2 ++
 .../templates/karavi-topology-service-account.yaml    |  2 ++
 .../templates/otel-collector.yaml                     | 11 +++++++++++
 6 files changed, 21 insertions(+)

diff --git a/charts/karavi-observability/templates/karavi-metrics-powerflex-service-account.yaml b/charts/karavi-observability/templates/karavi-metrics-powerflex-service-account.yaml
index 61f5478..9882026 100644
--- a/charts/karavi-observability/templates/karavi-metrics-powerflex-service-account.yaml
+++ b/charts/karavi-observability/templates/karavi-metrics-powerflex-service-account.yaml
@@ -5,6 +5,8 @@ kind: ServiceAccount
 metadata:
   name: {{ .Release.Name }}-metrics-powerflex-controller
   namespace: {{ include "custom.namespace" . }}
+imagePullSecrets:
+  - name: default-registry-key
 
 ---
 
diff --git a/charts/karavi-observability/templates/karavi-metrics-powermax-service-account.yaml b/charts/karavi-observability/templates/karavi-metrics-powermax-service-account.yaml
index 605c8df..f816bd6 100644
--- a/charts/karavi-observability/templates/karavi-metrics-powermax-service-account.yaml
+++ b/charts/karavi-observability/templates/karavi-metrics-powermax-service-account.yaml
@@ -5,6 +5,8 @@ kind: ServiceAccount
 metadata:
   name: {{ .Release.Name }}-metrics-powermax-controller
   namespace: {{ include "custom.namespace" . }}
+imagePullSecrets:
+  - name: default-registry-key
 
 ---
 
diff --git a/charts/karavi-observability/templates/karavi-metrics-powerscale-service-account.yaml b/charts/karavi-observability/templates/karavi-metrics-powerscale-service-account.yaml
index ee9c749..a2589a7 100644
--- a/charts/karavi-observability/templates/karavi-metrics-powerscale-service-account.yaml
+++ b/charts/karavi-observability/templates/karavi-metrics-powerscale-service-account.yaml
@@ -5,6 +5,8 @@ kind: ServiceAccount
 metadata:
   name: {{ .Release.Name }}-metrics-powerscale-controller
   namespace: {{ include "custom.namespace" . }}
+imagePullSecrets:
+  - name: default-registry-key
 
 ---
 
diff --git a/charts/karavi-observability/templates/karavi-metrics-powerstore-service-account.yaml b/charts/karavi-observability/templates/karavi-metrics-powerstore-service-account.yaml
index 7791dd2..d9e9f9b 100644
--- a/charts/karavi-observability/templates/karavi-metrics-powerstore-service-account.yaml
+++ b/charts/karavi-observability/templates/karavi-metrics-powerstore-service-account.yaml
@@ -5,6 +5,8 @@ kind: ServiceAccount
 metadata:
   name: {{ .Release.Name }}-metrics-powerstore-controller
   namespace: {{ include "custom.namespace" . }}
+imagePullSecrets:
+  - name: default-registry-key
 
 ---
 
diff --git a/charts/karavi-observability/templates/karavi-topology-service-account.yaml b/charts/karavi-observability/templates/karavi-topology-service-account.yaml
index 0c506ce..664a3e2 100644
--- a/charts/karavi-observability/templates/karavi-topology-service-account.yaml
+++ b/charts/karavi-observability/templates/karavi-topology-service-account.yaml
@@ -5,6 +5,8 @@ kind: ServiceAccount
 metadata:
   name: {{ .Release.Name }}-topology-controller
   namespace: {{ include "custom.namespace" . }}
+imagePullSecrets:
+  - name: default-registry-key
 
 ---
 
diff --git a/charts/karavi-observability/templates/otel-collector.yaml b/charts/karavi-observability/templates/otel-collector.yaml
index ca10c01..d1c400a 100644
--- a/charts/karavi-observability/templates/otel-collector.yaml
+++ b/charts/karavi-observability/templates/otel-collector.yaml
@@ -20,6 +20,16 @@ metadata:
 
 ---
 
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: otel-collector
+  namespace: {{ include "custom.namespace" . }}
+imagePullSecrets:
+  - name: default-registry-key
+
+---
+
 apiVersion: v1
 kind: Service
 metadata:
@@ -64,6 +74,7 @@ spec:
         app.kubernetes.io/name: otel-collector
         app.kubernetes.io/instance: {{ .Release.Name }}
     spec:
+      serviceAccount: otel-collector
       volumes:
       - name: tls-secret
         secret:
-- 
2.34.1
